src/shared/proto/messages_robocup_ssl_detection_tracked.proto

```diff
@@ -0,0 +1,122 @@
+syntax = "proto2";
+
+message Vector2 {
+    required float x = 1;
+    required float y = 2;
+}
+
+message Vector3 {
+    required float x = 1;
+    required float y = 2;
+    required float z = 3;
+}
+
+// The team color of the robot
+enum TeamColor {
+    UNKNOWN = 1;
+    YELLOW = 2;
+    BLUE = 3;
+}
+
+message RobotId {
+    // The robot number
+    required uint32 id = 1;
+    // The team color
+    required TeamColor team_color = 2;
+}
+
+enum ChipKickDetectionSupport {
+    UNSUPPORTED = 0;
+    FIRST_TOUCHDOWN_ONLY = 1;
+    ALL_TOUCHDOWNS = 2;
+}
+
+// A single tracked ball
+message TrackedBall {
+    // The quality of the detection
+    // A value between 0 (not visible) and 1 (visible)
+    // The exact implementation depends on the source software
+    optional float quality = 1;
+
+    // The position (x, y, height) [m] in the ssl-vision coordinate system
+    required Vector3 pos = 2;
+
+    // The velocity [m/s] in the ssl-vision coordinate system
+    optional Vector3 vel = 3;
+
+    // The level of chip kick detection support by the source implementation
+    required ChipKickDetectionSupport chip_kick_detection_support = 4;
+
+    // The touch down location(s), if the ball is flying.
+    // Depending on the source implementation, the next or all touch down locations may be provided
+    // ChipKickDetectionSupport#UNSUPPORTED -> there are no valid touch down locations
+    // ChipKickDetectionSupport#FIRST_TOUCHDOWN_ONLY -> there is only one touch down location, although the ball may have more
+    // ChipKickDetectionSupport#ALL_TOUCHDOWNS -> all touch down locations are included
+    repeated Vector2 touch_down_locations = 5;
+}
+
+// A ball kicked by a robot, including predictions when the ball will come to a stop
+message KickedBall {
+    // The initial position [m] from which the ball was kicked
+    required Vector3 pos = 1;
+    // The initial velocity [m/s] with which the ball was kicked
+    required Vector3 vel = 2;
+    // The unix timestamp [s] when the kick was performed
+    required double timestamp = 3;
+
+    // The predicted time [s] that ball ball needs to come to a stop
+    optional float stop_time = 4;
+    // The predicted position [m] at which the ball will come to a stop
+    optional Vector2 stop_pos = 5;
+
+    // The robot that kicked the ball
+    optional RobotId robot_id = 6;
+
+    // A flag to indicate if the ball was chipped initially
+    optional bool chipped = 7;
```

Comment:
This is redundant and can be removed. An initial Z velocity != 0 indicates a chip kick.